File: backend/src/main/java/com/cdcdemo/dto/CdcEvent.java
package com.cdcdemo.dto;

import java.util.Map;
import java.util.UUID;

public class CdcEvent {

    private String eventId;
    private String operation;   // INSERT, UPDATE, DELETE, SNAPSHOT
    private String tableName;
    private long timestamp;
    private String lsn;
    private Long txId;
    private Map<String, Object> before;
    private Map<String, Object> after;

    public CdcEvent() {
        this.eventId = UUID.randomUUID().toString();
    }

    public String getEventId() { return eventId; }
    public void setEventId(String eventId) { this.eventId = eventId; }

    public String getOperation() { return operation; }
    public void setOperation(String operation) { this.operation = operation; }

    public String getTableName() { return tableName; }
    public void setTableName(String tableName) { this.tableName = tableName; }

    public long getTimestamp() { return timestamp; }
    public void setTimestamp(long timestamp) { this.timestamp = timestamp; }

    public String getLsn() { return lsn; }
    public void setLsn(String lsn) { this.lsn = lsn; }

    public Long getTxId() { return txId; }
    public void setTxId(Long txId) { this.txId = txId; }

    public Map<String, Object> getBefore() { return before; }
    public void setBefore(Map<String, Object> before) { this.before = before; }

    public Map<String, Object> getAfter() { return after; }
    public void setAfter(Map<String, Object> after) { this.after = after; }
}

File: frontend/src/types/CdcEvent.ts
export type Operation = 'INSERT' | 'UPDATE' | 'DELETE' | 'SNAPSHOT'

export interface CdcEvent {
  eventId: string
  operation: Operation
  tableName: string
  timestamp: number
  lsn: string | null
  txId: number | null
  before: Record<string, unknown> | null
  after: Record<string, unknown> | null
}

export interface Product {
  id: number
  name: string
  category: string
  price: number
  stock: number
  createdAt: string
  updatedAt: string
}

export interface ProductForm {
  name: string
  category: string
  price: string
  stock: string
}
